test/webdav_test.dart

```diff
@@ -1,88 +1,15 @@
 import 'dart:convert';
 import 'dart:io';
 
-import 'package:browser_launcher/browser_launcher.dart';
 import 'package:nextcloud/nextcloud.dart';
 import 'package:test/test.dart';
 
-class Config {
-  const Config({
-    this.host,
-    this.username,
-    this.password,
-    this.shareUser,
-    this.testDir,
-    this.email,
-    this.storageLocation,
-  });
-
-  factory Config.fromJson(Map<String, dynamic> json) => Config(
-        host: Uri.parse(json['host']),
-        username: json['username'],
-        password: json['password'],
-        shareUser: json['shareUser'],
-        // normalised path (remove trailing slash)
-        testDir: json['testDir'].endsWith('/')
-            ? json['testDir'].substring(0, json['testDir'].length - 1)
-            : json['testDir'],
-        email: json['email'],
-        storageLocation: json['storageLocation'],
-      );
+import 'config.dart';
 
-  final Uri host;
-  final String username;
-  final String password;
-  final String shareUser;
-  final String testDir;
-  final String email;
-  final String storageLocation;
-}
-
-// TODO: add more tests
 void main() {
```

Comment:
`test('Clean test environment', () async {` this test now breaks the tests in `preview_test.dart` when running all tests, because of parallel execution. Probably best to split up into separate test directories where every test class is self responsible for setup and clean up.

Reply:
I didn't hit any of those race conditions, but separating makes sense